### PR TITLE
Backend implementation of clustering feature

### PR DESCRIPTION
**Pull request by nandeeshp24**
* Dialogflow integration to get parameter values for clustering from the user
* Clustering functionality based on the parameter values received from the user

### CODE REVIEW

File: backend-server/src/main/java/org/dice/ida/action/def/ClusterAction.java [hunk 1]
```diff
@@ -0,0 +1,417 @@
+package org.dice.ida.action.def;
```
Comment (nikit91) [inline backend-server/src/main/java/org/dice/ida/action/def/ClusterAction.java:1]: Missing method level comments

File: backend-server/src/main/java/org/dice/ida/model/clustering/FarthestFirstAttribute.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package org.dice.ida.model.clustering;
+
+import weka.clusterers.FarthestFirst;
+
+public class FarthestFirstAttribute {
```
Comment (nikit91) [inline backend-server/src/main/java/org/dice/ida/model/clustering/FarthestFirstAttribute.java:5]: Please add javadoc comments describing the class and its methods

File: backend-server/src/main/java/org/dice/ida/model/clustering/KmeansAttribute.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+	private int NumberOfCluster;
+	private String IntitializeMethod;
+	private int MaxIterations;
+	private Boolean ReplaceMissingValues;
+	private int NumOfExecutionSlots;
+	private int RandomNumberSeed;
+	private SimpleKMeans simpleKMeans;
+
+	public KmeansAttribute(SimpleKMeans simpleKMeans, int numcluster)
```
Comment (nikit91) [inline backend-server/src/main/java/org/dice/ida/model/clustering/KmeansAttribute.java:16]: Please add javadoc comments describing the class and its methods

**Review by nikit91 — CHANGES_REQUESTED** [review]
Please provide class and method level comments

**Comment by sourabhpoddar404** [comment]
Java Doc Comments added

**Review by nikit91 — APPROVED** [review]
lgtm

**Pull request merged**